assets/training/finetune_acft_hf_nlp/src/model_selector/model_selector.py

```diff
@@ -331,8 +298,57 @@ def get_parser():
     return parser
 
 
+def is_model_folder_input(args: Namespace) -> bool:
+    """Check if model folder is passed as input"""
+    return any(
+        [
+            getattr(args, "pytorch_model_path", None),
+            getattr(args, "mlflow_model_path", None)
+        ]
+    )
+
+
+def load_ft_config(args: Namespace):
+    """Search for finetune config and update ACFT_CONFIG.
+
+    preference: user passed finetune config > model packaged finetune config.
+    """
+    global ACFT_CONFIG
+
+    # user passed finetune config
+    # check if finetune config is inside pytorch/mlflow model folder
+    if (
+        args.finetune_config_path is None and
+        is_model_folder_input(args)
+    ):
+        model_folder_input = (
+            args.pytorch_model_path
+            if getattr(args, "pytorch_model_path", None) is not None else
+            args.mlflow_model_path
+        )
+        logger.info(f"Model folder input: {model_folder_input}")
+        finetune_config_path = Path(model_folder_input, SaveFileConstants.ACFT_CONFIG_SAVE_PATH)
+        if finetune_config_path.is_file():
+            logger.info(f"Found finetune config path in model artifacts folder: {finetune_config_path}.")
+            setattr(args, "finetune_config_path", str(finetune_config_path))
+
```

Comment:
The finetune_config file in base model will be of same structure as what we output. The structure will have 1 level less. It will contain only details regarding the current model.